src/cli/cli.jl

```diff
@@ -9,22 +9,27 @@ module CLI
 
 const OptsType = Base.Dict{Base.Symbol,Base.Any}
 
-using OpenPolicyAgent_jll
-
 """
 CommandLine execution context.
 
-`exec`: a no argument function that provides the base command to execute in a julia `do` block.
-`cmdopts`: keyword arguments that should be used to further customize the `Cmd` creation
-`pipelineopts`: keyword arguments that should be used to further customize the `pipeline` creation
+CommandLine(exec; cmdopts, pipelineopts, runopts)
+
+- `exec`: a no argument function that provides the base command to execute in a julia `do` block.
+- `cmdopts`: keyword arguments that should be used to further customize the `Cmd` creation
+- `pipelineopts`: keyword arguments that should be used to further customize the `pipeline` creation
+- `runopts`: additional options to be passed to the `Base.run` method
 """
 Base.@kwdef struct CommandLine
-    exec::Base.Function = OpenPolicyAgent_jll.opa
+    exec::Base.Function = ()->``
     cmdopts::OptsType = OptsType()
     pipelineopts::OptsType = OptsType()
     runopts::OptsType = OptsType()
 end
 
+function CommandLine(exec::Base.Function; kwargs...)
+    return CommandLine(; exec=exec, kwargs...)
+end
```

Comment:
Do you expect the users to use the 4-keyword constructor? If not, then this should probably be an inner constructor? Defaulting to an empty command seems a bit weird. I.e. something like

```julia
struct CommandLine
    exec::Base.Function
    cmdopts::OptsType
    pipelineopts::OptsType
    runopts::OptsType

    function CommandLine(exec::Base.Function; cmdopts::OptsType = OptsType(), pipelineopts::OptsType = OptsType(), runopts::OptsType = OptsType())
        return new(exec, cmdopts, pipelineopts, runopts)
    end
end
```
?

Reply:
Yes... let's change it

Reply:
Done in https://github.com/JuliaComputing/OpenPolicyAgent.jl/pull/17/commits/7f777bd21ddb246657d3f21db7458b46e5f4ee0b